From 801ef648e22f997e0cc7e8895d0c917af5a3bae5 Mon Sep 17 00:00:00 2001
From: Alfred Chen <cchalpha@gmail.com>
Date: Tue, 21 Sep 2021 11:14:18 +0000
Subject: [PATCH 214/232] sched/alt: Fix missing cpuacct.usage.

Call to cgroup_account_cputime() is missing in Project C scheduler,
which cause cpuacct.usage reports zero.

This should fixed #40
---
 kernel/sched/alt_core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/sched/alt_core.c b/kernel/sched/alt_core.c
index 3d334abeadce..c1643fa8243c 100644
--- a/kernel/sched/alt_core.c
+++ b/kernel/sched/alt_core.c
@@ -3468,6 +3468,7 @@ static inline void update_curr(struct rq *rq, struct task_struct *p)
 	s64 ns = rq->clock_task - p->last_ran;
 
 	p->sched_time += ns;
+	cgroup_account_cputime(p, ns);
 	account_group_exec_runtime(p, ns);
 
 	p->time_slice -= ns;
-- 
2.33.1.711.g9d530dc002

